packages/driver/src/cy/commands/actions/check.ts

```diff
@@ -144,6 +144,10 @@ const checkOrUncheck = (Cypress, cy, type, subject, values = [], userOptions = {
           options._log.snapshot().end()
         }
 
+        if ($el.prop('intermediate')) {
```

Comment:
Pretty sure that's not the right place (or even the right change) for the changes but this is as far to where my tracing has led. I'm also not sure if this solves the problem as I have no idea how to test it.

Reply:
You've found *almost* the right place - right file, right block, just needs to be above `snapshot().end()` conditional - we'd want to remove the property before taking an "end state" DOM snapshot.

As far as testing it, for manually testing it, you'd want to clone the cypress repo and run `yarn cypress:open` - you could then test your project where you're using indeterminate checkboxes.

For automated testing, which we'll need before merging the PR, you'd add a new test to https://github.com/cypress-io/cypress/blob/develop/packages/driver/cypress/integration/commands/actions/check_spec.js#L72. It would be fairly simple, just a new test that adds `indeterminate` to a checkbox, `.check()`s it, then verifies that indeterminate is gone and the box is checked. Similarly for https://github.com/cypress-io/cypress/blob/develop/packages/driver/cypress/integration/commands/actions/check_spec.js#L852, add indeterminate to a checkbox, `uncheck()` it, then verify that indeterminate is gone and the box is unchecked.

Hopefully that's helpful, thanks for being patient while we were all out on vacation. :)